gender_novels/analysis/visualizations/datagraphs_functions.py

```diff
@@ -2,58 +2,87 @@
+    plt.yticks(size=11,color='slategray')
+    plt.xticks([i for i in range(min(years),max(years)+9,10)],size=11,color='slategray')
     for label in ax1.xaxis.get_ticklabels():
-        label.set_rotation(45)
-    plt.show()
+        label.set_rotation(60)
+    plt.subplots_adjust(left=.1,bottom=.18,right=.95,top=.9)
+    plt.savefig('date_of_pub_for_'+name_of_data+'.png')
 
-def plt_pubcountries(pub_country):
+def plt_pubcountries(pub_country,name_of_data):
     '''
     Creates a bar graph displaying the frequency of books that were published in each country
     :param pub_country: list
-    :return:
+    RETURNS a pyplot bargraph
     '''
-    sns.set(style='darkgrid')
+    sns.set_style('darkgrid')
     ax1=plt.subplot2grid((1,1),(0,0))
     country_counter={}
-    x=[]
-    y=[]
     for country in pub_country:
-        if country in country_counter:
-            country_counter[country]+=1
-        else:
-            country_counter[country]=1
-    for i in country_counter:
-        x.append(i)
-        y.append(country_counter[i])
-
+        country_counter[country]=country_counter.setdefault(country,0)+1
+    x=[country for country in country_counter]
+    y=[country_counter[key] for key in country_counter]
     for label in ax1.xaxis.get_ticklabels():
-        label.set_rotation(45)
-    plt.bar(x,y,color='red')
-    plt.xlabel('Countries')
-    plt.ylabel('Frequency')
-    plt.title('Country of Publication')
-    plt.show()
+        label.set_rotation(15)
+    plt.bar(x,y,color='plum')
+    plt.xlabel('Countries',size=13,weight='bold',color='slategray')
+    plt.ylabel('Frequency',size=13,weight='bold',color='slategray')
+    plt.title('Country of Publication for '+name_of_data,size=15,color='slategray',weight='bold')
+    plt.xticks(color='slategray',size=12)
+    plt.yticks(color='slategray',size=12)
+    plt.subplots_adjust(left=.1,bottom=.18,right=.95,top=.9)
+    plt.savefig('country_of_pub_for_'+name_of_data+'.png')
+
+def plt_gender_breakdown(pub_gender,name_of_data):
+    '''
+    Creates a pie chart displaying the composition of male and female writers in the data
+    :param pub_gender: list
+    :param name_of_data: str
+    RETURNS a pie chart
+    '''
+    gendercount={}
+    for i in pub_gender:
+        gendercount[i]=gendercount.setdefault(i,0)+1
+    total=0
+    for i in gendercount:
+        total+=gendercount[i]
+    slices=[gendercount[i]/total for i in gendercount]
+    genders=[i for i in gendercount]
+    labelgenders=[]
+    for i in range(len(genders)):
+        labelgenders.append(genders[i]+': ' + str(round(slices[i],2)*100)+'%')
+    colors=['slateblue','mediumpurple','plum']
+    plt.pie(slices,colors=colors,labels=labelgenders)
+    plt.title('Gender Breakdown for '+name_of_data,size=15,color='slategray',weight='bold')
+    plt.legend()
+    plt.subplots_adjust(left=.1,bottom=.1,right=.9,top=.9)
+    plt.savefig('gender_breakdown_for_'+name_of_data+'.png')
+
+if __name__=='__main__':
+    pubyears=[]
```

Comment:
can you take all of the code from here and put it into a `create_corpus_summary_visualizations(corpus_name)` function? 
That way, another team member can just call 
```
create_corpus_summary_visualizations('sample_novels')
create_corpus_summary_visualizations('gutenberg')
```
to update all of our corpus visualizations.
```suggestion
    pubyears=[]
```